django/db/models/expressions.py

```diff
@@ -400,7 +429,7 @@ def identity(self):
             identity.append((arg, value))
         return tuple(identity)
 
-    def __eq__(self, other):
```

Comment:
Just to show this proof of operation, I've renamed this method. I'm not sure what we'd need to do about a deprecation process for this, or if this makes the whole process a washout.